QuizApp/Client/Account/AccountNetworkClient.swift

```diff
@@ -0,0 +1,26 @@
+import Foundation
+
+class AccountNetworkClient: AccountNetworkClientProtocol {
+
+    private let baseNetworkClient: BaseNetworkClientProtocol
+
+    init(baseNetworkClient: BaseNetworkClientProtocol) {
+        self.baseNetworkClient = baseNetworkClient
+    }
+
+    func getAccountDetails() async throws -> AccountDetailsResponse {
+        try await baseNetworkClient.get(
+            url: AccountEndpoints.account.path,
+            params: [:],
+            headers: [HeaderField.contentType.key: HeaderValue.defaultContentType.value])
+    }
+
+    func updateUsername(requestModel: UsernameUpdateRequestModel) async throws -> AccountDetailsResponse {
```

Comment:
This is misleading. The API allows you to update any account field, except the username. For now, it is only a name, but that could expand in the future. What if I add another field?
More proper name would be `func update(request: AccountUpdateRequestModel)`